Marlin/src/lcd/ultralcd.cpp

```diff
@@ -1092,17 +1092,16 @@ void lcd_quick_feedback(const bool clear_buttons) {
     static void IDEX_menu() {
       START_MENU();
       MENU_BACK(MSG_MAIN);
-
-      MENU_ITEM(gcode, MSG_IDEX_MODE_AUTOPARK,  PSTR("M605 S1\nG28 X\nG1 X100"));
+      char menu_buffer[65];
+      sprintf_P(menu_buffer, PSTR("M605 S1\nG28 X\nG1 X%i"), int(X_BED_SIZE / 2));
+      MENU_ITEM(gcode, MSG_IDEX_MODE_AUTOPARK, menu_buffer);
       const bool need_g28 = !(TEST(axis_known_position, Y_AXIS) && TEST(axis_known_position, Z_AXIS));
-      MENU_ITEM(gcode, MSG_IDEX_MODE_DUPLICATE, need_g28
-        ? PSTR("M605 S1\nT0\nG28\nM605 S2 X200\nG28 X\nG1 X100")                // If Y or Z is not homed, do a full G28 first
-        : PSTR("M605 S1\nT0\nM605 S2 X200\nG28 X\nG1 X100")
-      );
-      //MENU_ITEM(gcode, MSG_IDEX_MODE_SCALED_COPY, need_g28
-      //  ? PSTR("M605 S1\nT0\nG28\nM605 S2 X200\nG28 X\nG1 X100\nM605 S3 X200")  // If Y or Z is not homed, do a full G28 first
-      //  : PSTR("M605 S1\nT0\nM605 S2 X200\nG28 X\nG1 X100\nM605 S3 X200")
-      //);
+      const char * const g28 = need_g28 ? "G28\n" : "";
+      sprintf_P(menu_buffer, PSTR("M605 S1\nT0\n%sM605 S2 X%i\nG28 X\nG1 X%i"), g28, DEFAULT_DUPLICATION_X_OFFSET, int(X_BED_SIZE / 3));
+      MENU_ITEM(gcode, MSG_IDEX_MODE_DUPLICATE, menu_buffer);
+      sprintf_P(menu_buffer, PSTR("M605 S1\nT0\n%sM605 S2 X%i\nG28 X\nG1 X%i\nM605 S3 X%i"), g28, DEFAULT_DUPLICATION_X_OFFSET, int(X_BED_SIZE / 3), DEFAULT_DUPLICATION_X_OFFSET);
+      //MENU_ITEM(gcode, MSG_IDEX_MODE_SCALED_COPY, menu_buffer);
```

Comment:
None of this will work. `MENU_ITEM(gcode, ...)` requires a PROGMEM string.

Reply:
Based on what you said before, I just started changing these to use a function callback to enqueue_and_echo_command_now - that should do the job!

Reply:
You'll have to add to the menu handler:
```cpp
if (no_reentry) return;
```
…and when sending any command…
```cpp
no_reentry = true;
```
…and after sending any command…
```cpp
no_reentry = false;
```

Reply:
You'll have to add to the menu handler:
```cpp
if (no_reentry) return;
```
…and call `lcd_enqueue_command` instead of `enqueue_and_echo_command_now`.

Reply:
Note that during `G28` the menu will be unresponsive. For this reason we've done things like `_lcd_level_bed_homing`.

Reply:
already found the lcd_enqueue command a few minutes ago! Ill look at the lcd_level_bed_homing function next